clients/fides-js/src/fides.ts

```diff
@@ -131,26 +133,40 @@ const automaticallyApplyGPCPreferences = (
   fidesApiUrl: string,
   effectiveExperience?: PrivacyExperience | null
 ) => {
-  if (!effectiveExperience) {
+  if (!effectiveExperience || !effectiveExperience.privacy_notices) {
     return;
   }
 
-  if (!getConsentContext().globalPrivacyControl) {
+  const context = getConsentContext();
+  if (!context.globalPrivacyControl) {
     return;
   }
 
-  const consentPreferencesToSave: Array<SaveConsentPreference> = [];
```

Comment:
this change fixes a pretty subtle bug, where:
1. let's say there are two notices, one notice-only `first` and one opt-out `second`
2. when fides-js first initializes, it figures out to make the cookie `{first: false, second: true}`
3. Then in this gpc function, because this originally only pushed to `consentPreferencesToSave` if certain conditions were met (`has_gpc_flag` and no current preference), `consentPreferencesToSave` often would not have all the keys. In this case, only `second` would be pushed, since `first` is `notice_only` so would have `has_gpc_flag=false`
4. It would send `consentPreferencesToSave` to `updateConsentPreferences` which sends the update to the backend. However, it also updates the cookie based on this value, so the cookie goes from `{first: false, second: true}` --> `{second: false}`, when really we want `{first: false, second: false}`. This resulted in the notice-only notice looking like it was always opted out!

Instead, we make sure we always map over all privacy notices here, and only set to `false` in certain cases. the important part is that `updateConsentPreferences` gets _all_ the relevant preferences